### PR TITLE
Miho and ammo

### PR DESCRIPTION
**Pull request by Tostov**
## Additions

Non-Mod Specific nailgun and HV nailgun, as well as ammo for 47x398mmR and 57x280mmR shells.
Miho specific ammunition for the aformentioned cannon shells (race specific AP-I for the 47, race specific canister and thermobaric for the 57), as well as a few other calibers they use (AP-I variants).
https://docs.google.com/spreadsheets/d/1MTHK21vMpdeQG12cwJu8P3bDIndNRs_l4OVt1EmSE_o/edit?usp=sharing

Patches for the new stuff, incl. rocket launcher which while guided in vanilla we cannot do yet, so it's dumbfire for the moment.

## Changes

Caliber changes for a number of weapons based on descriptions. (40mm grenade to 20mm grenade, 50mm rocket to 57mm rocket). Changed the 57mm grenade launcher for a 57mm cannon that it is actually supposed to be. The 100mm incendiary rockets are now explosive/thermobaric as their vanilla performance is.

Removed patches for stuff that is no longer in the mod.

## Reasoning

The 57mm gun is pretty obviously supposed to be the other jap tank cannon after a bit of poking around (https://en.wikipedia.org/wiki/Type_97_Chi-Ha_medium_tank the 57 and 47 are armaments for this tank). The AP-I, thermobaric and canister shots for the cannons are left to being mod specific because none of them were actual loads for the guns.

The 13.2mm HMG caliber is probably supposed to be 13.2×99mm Hotchkiss rather than 13.2mm TuF, but the two calibers are probably extremely close in spec (51.5@785 vs 52@790), and we don't actually have 13.2x99mm.

## Alternatives

We could add the other calibers to the generics (with regular damage). It's probably not entirely unreasonable to give both the 47/57mm guns APHE/AP-I rounds either but falls outside of kind of the main scope of adding the base shells.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [ ] Playtested a colony (specify how long)

### CODE REVIEW

**Comment by N7Huntsman** [comment]
Have come to the following decision regarding the new ammo types:
- **Nails:** As much as it feels _genuinely wrong_, I have to agree they're a generic enough "ammo" to warrant adding to core. I've added them to sheet and am hoping to get some better textures for them. For now, I've held off on converting the Rimsenal weaponry to the "new" nails, pending discussion.
- **57mm and 47mm cannon:** Both are very niche calibers that were only ever used by a Japan and in comparatively limited numbers. If we ever suddenly patch a gluttony of WW2 Japanese vehicles/emplacement mods we can reassess, but for now I've moved them over to the Miho patch.

File: ModPatches/Miho Race/Patches/Miho Race/ThingDefs_Misc/Weapons_Other.xml [hunk 1]
```diff
@@ -1,4 +1,112 @@
...
+
+	<Operation Class="PatchOperationReplace">
+		<xpath>Defs/ThingDef[
+			defName="Miho_Turret_Missile"
+			]/building/turretBurstCooldownTime </xpath>
+		<value>
+			<turretBurstCooldownTime>1.5</turretBurstCooldownTime>
+		</value>
+	</Operation>
+	<!--Temp ammo for testing-->
```
Comment (N7Huntsman) [inline ModPatches/Miho Race/Patches/Miho Race/ThingDefs_Misc/Weapons_Other.xml:75]: Should this have a different ammo type?

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
No errors in start up, but see my remarks.

File: ModPatches/Miho Race/Patches/Miho Race/ThingDefs_Misc/Weapons_Other.xml (hunk 1 above)
Comment (Tostov) [reply ModPatches/Miho Race/Patches/Miho Race/ThingDefs_Misc/Weapons_Other.xml:75]: Ahh no it did have RPG-7 ammo initially, forgot to remove the comment when I actually updated it to the correct ammo.

Comment (Tostov) [reply ModPatches/Miho Race/Patches/Miho Race/ThingDefs_Misc/Weapons_Other.xml]: And it turns out I forgot to update default projectile again... Should be resolved now.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**